src/main/java/com/edmunds/rest/databricks/service/JobServiceImpl.java

```diff
@@ -146,8 +151,14 @@ public List<JobDTO> getJobsByRegex(Pattern regex) throws IOException, Databricks
   }
 
   @Override
-  public JobsDTO listAllJobs() throws DatabricksRestException, IOException {
-    byte[] responseBody = client.performQuery(RequestMethod.GET, "/jobs/list", null);
+  public JobsDTO listAllJobs(int limit, int offset, boolean expandTasks) throws DatabricksRestException, IOException {
```

Comment:
so this will still return all jobs? The limit parameter is just for runs? I saw the javadoc too just confirming.

Reply:
hmm... from what i see it limits the number of jobs, not job runs. Quoting: 

The number of jobs to return. This value must be greater than 0 and less or equal to 25


Databricks changed this API in 2.1... Do you want to change the name to "listJobs" instead of "listAllJobs" or introduce additional logic to iteratively load all jobs inside the lib (I would vote against it)

Reply:
oh I see. Given that this will not be backwards compatible, changing the method name to listJobs sounds like a good idea.

Reply:
@samshuster I just realized that if we do not implement `listAllJobs` then `findJobByName` and `findJobByRegex` will work incorrectly :( Will need time to implement this one.

P.S. Renamed

Reply:
oh right ok. Perhaps it is best to take a step back and think about making a brand new interface for this new job service? This way you can decide what is best to implement as part of it given the new api definition/constraints. 

We can then still keep the old implementation for this initial release but mark it as deprecated with a future planned removal.

What do you think?

Reply:
I was thinking the first one. Basically two different JobServices with two completely different interfaces. The databricksServiceFactory could have a method for both legacy jobservice and the new job service.

This way for the new job service, the interface could be modified to no longer require some of these problematic methods.

This would also make it easier to roll out this change without worrying so much about large migration efforts immediately.

Reply:
Got it..... 
Do you have a preferred name for the new interface/methods? 
Also, do you still want to remove the `withApi` functions? If yes - what is the meaning of it?

Reply:
Hmm naming is tricky. I guess I would lean towards naming the new service and impl something like:
- JobServiceV21
What are your thoughts on that?

And sorry could you elaborate on the removal of the withApi functions comment? I can't remember what this is referring to